trewrap/src/test/java/org/codice/imaging/nitf/trewrap/SECURA_WrapTest.java

```diff
@@ -0,0 +1,261 @@
+        assertNull(securityMetadata.getClassificationText());
+        assertNull(securityMetadata.getClassificationAuthorityType());
+        assertEquals("", securityMetadata.getClassificationAuthority());
+        assertNull(securityMetadata.getClassificationReason());
+        assertEquals("", securityMetadata.getSecurityControlNumber());
+    }
+
+    public static void checkNitf21SecurityMetadataUnclasAndEmpty(SecurityMetadata securityMetadata) {
+        Assert.assertEquals(SecurityClassification.UNKNOWN, securityMetadata.getSecurityClassification());
+        assertEquals("", securityMetadata.getSecurityClassificationSystem());
+        assertEquals("", securityMetadata.getCodewords());
+        assertEquals("", securityMetadata.getControlAndHandling());
+        assertEquals("", securityMetadata.getReleaseInstructions());
+        assertEquals("", securityMetadata.getDeclassificationType());
+        assertEquals("", securityMetadata.getDeclassificationDate());
+        assertEquals("", securityMetadata.getDeclassificationExemption());
+        assertEquals("", securityMetadata.getDowngrade());
+        assertEquals("", securityMetadata.getDowngradeDate());
+        assertEquals("", securityMetadata.getClassificationText());
+        assertEquals("", securityMetadata.getClassificationAuthorityType());
+        assertEquals("", securityMetadata.getClassificationAuthority());
+        assertEquals("", securityMetadata.getClassificationReason());
+        assertEquals("", securityMetadata.getSecurityControlNumber());
+    }
+
+    private static byte[] compress(String data) {
+        byte[] compressed = null;
+        try (ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        GZIPOutputStream gzip = new GZIPOutputStream(bos);) {
+            gzip.write(data.getBytes(StandardCharsets.ISO_8859_1));
+            // in order to get the full set of bytes you must close the output stream.
+            gzip.close();
+            compressed = bos.toByteArray();
+        } catch (Exception ex) {
+            System.out.println(ex.getLocalizedMessage());
+        }
+        return compressed;
+    }
+
+    private final String uncompress(byte[] compressed) throws NitfFormatException {
+            StringBuilder sb = new StringBuilder();
+            try (ByteArrayInputStream bis = new ByteArrayInputStream(compressed);
+                    GZIPInputStream gis = new GZIPInputStream(bis);
+                    BufferedReader br = new BufferedReader(new InputStreamReader(gis));) {
+                String line;
+                while((line = br.readLine()) != null) {
+                    sb.append(line);
+                }
+            } catch (IOException e) {
+                throw new NitfFormatException("Could not uncompress SECURITY field.");
+            }
+            return sb.toString();
+    }
+
+    private final byte[] buildTREBytes() {
+        String result;
+        String defaltEncoding = System.getProperty("file.encoding");
```

Comment:
default (typo)

I'm really not sure this use of `file.encoding` is ever a good idea - your test results depend on the locale of the machine where its being run.

Reply:
retyped.

Reply:
Ok, correct me if I'm wrong here.  

I was wrong, just did some more reading up on Java string/character encoding.

Changed to UTF-8.